crates/p2p/src/tests/request_response.rs

```diff
@@ -1,50 +1,84 @@
+    } = Setup::all_to_all(2).await?;
 
+    let req_msg = b"request from node1".to_vec();
+    let resp_msg = b"response from node2".to_vec();
     user_handles[0]
-        .handle
+        .command
         .send_command(Command::RequestMessage {
             peer_id: user_handles[1].peer_id,
-            data: Vec::<u8>::from("Hello, it's a request..."),
+            data: req_msg.clone(),
         })
         .await;
+    info!("Node 1 sent request to Node 2");
+
+    #[cfg(feature = "request-response")]
+    {
+        match user_handles[1].reqresp.next_event().await.unwrap() {
+            ReqRespEvent::ReceivedRequest(data, channel) => {
+                info!("Node 2 received request: {:?}", data);
+                assert_eq!(data, req_msg, "Node 2 did not receive the correct request");
+                let _ = channel.send(resp_msg.clone());
+                info!("Node 2 sent response");
+            }
+            _ => unreachable!("Node 2 did not receive a request"),
+        }
 
-    match user_handles[1].handle.next_event().await {
-        Ok(event) => match event {
-            Event::ReceivedMessage(_data) => {
-                bail!(
-                    "Something is insanely wrong: it got a gossipsub's message when it should have got a request from Request-response protocol.",
+        match user_handles[0].reqresp.next_event().await.unwrap() {
+            ReqRespEvent::ReceivedResponse(resp) => {
+                info!("Node 1 received response: {:?}", resp);
+                assert_eq!(
+                    resp, resp_msg,
+                    "Node 1 did not receive the correct response",
                 );
             }
-            Event::ReceivedRequest(data) => {
-                assert_eq!(data, Vec::<u8>::from("Hello, it's a request..."));
+            _ => unreachable!("Node 1 did not receive a response"),
+        }
+    }
+
+    #[cfg(not(feature = "request-response"))]
+    {
+        match user_handles[1].reqresp.next_event().await.unwrap() {
+            ReqRespEvent::ReceivedRequest(data) => {
+                info!("Node 2 received request: {:?}", data);
+                assert_eq!(data, req_msg, "Node 2 did not receive the correct request");
+
+                user_handles[1]
+                    .command
+                    .send_command(Command::PublishMessage {
+                        data: resp_msg.clone(),
+                    })
+                    .await;
+                info!("Node 2 sent response via gossip");
             }
-        },
-        Err(e) => bail!("Something is wrong: {e}"),
+            _ => unreachable!("Node 2 did not receive a request"),
+        }
+
+        match user_handles[0].gossip.next_event().await {
+            Ok(event) => match event {
+                GossipEvent::ReceivedMessage(data) => {
+                    assert_eq!(data, resp_msg);
+                }
+            },
+            Err(e) => info!("Something is wrong: {:?}", e),
+        }
```

Comment:
Same: why do we need to support req-resp things when the feature is not enabled?